librz/include/rz_bin_dwarf.h

```diff
@@ -698,9 +1046,9 @@ typedef enum {
 	DW_AT_KIND_STRING,
 } RzBinDwarfAttrKind;
 
-typedef struct dwarf_attr_kind {
-	ut64 attr_name;
-	ut64 attr_form;
+typedef struct dwarf_attr_t {
+	enum DW_AT name;
```

Comment:
Is this really the `name` of the attribute? Because there is also an `DW_AT_name` entry in the enum. Isn't this more the type? Sorry. I have no real experience with dwarf.

Reply:
It is the DW_AT property of this dwarf_attr_t.

```
[2] DW_TAG_subprogram DW_CHILDREN_yes
	DW_AT_low_pc DW_FORM_addr
	DW_AT_high_pc DW_FORM_data4
	DW_AT_frame_base DW_FORM_exprloc
	DW_AT_name DW_FORM_strp
	DW_AT_decl_file DW_FORM_data1
	DW_AT_decl_line DW_FORM_data1
	DW_AT_type DW_FORM_ref4
	DW_AT_external DW_FORM_flag_present
```

Like this DIE, it has a lot of DW_AT_XX DW_FORM_XX pairs, each described by a dwarf_attr_t.

Then the name is actually the name from the old code, I just didn't feel like changing it.